core/src/main/java/org/elasticsearch/index/query/QueryShardContext.java

```diff
@@ -73,15 +68,13 @@
 /**
  * Context object used to create lucene queries on the shard level.
  */
-public class QueryShardContext {
+public class QueryShardContext extends QueryRewriteContext {
```

Comment:
Sure I see where this is handy, but from a design perspective, does the `QueryShardContext` ever need to parse anymore? or does it need the ScriptService? I think it may be convenient to isolate those to make sure that `toQuery` doesn't do any parsing/scripting at this point?